src/DotNetWorker.Core/Converters/Converter/DefaultInputConverterProvider.cs

```diff
@@ -50,12 +50,30 @@ public IEnumerable<IInputConverter> RegisteredInputConverters
                 {
                     yield return _converterCache.GetOrAdd(converterType.AssemblyQualifiedName!, (key) =>
                     {
-                        return (IInputConverter)ActivatorUtilities.CreateInstance(_serviceProvider, converterType);
+                        return GetOrCreateConverterInstance(converterType);
                     });
                 }
             }
         }
 
+        /// <summary>
+        /// Gets an instance of the converter for the type requested.
+        /// </summary>
+        /// <param name="converterType">The type for which we are requesting an IInputConverter instance.</param>
+        /// <exception cref="InvalidOperationException">Throws when the converterType param is null.</exception>
+        /// <returns>IConverter instance of the requested type.</returns>
+        public IInputConverter GetOrCreateConverterInstance(Type converterType)
+        {
+            if (converterType is null)
+            {
+                throw new InvalidOperationException($"Could not create an instance of {(nameof(converterType))}.");
+            }
+
+            EnsureTypeCanBeAssigned(converterType);
+
+            return (IInputConverter)ActivatorUtilities.CreateInstance(_serviceProvider, converterType);
```

Comment:
@surgupta-msft, what I was meaning is this change:
```suggestion
            return (IInputConverter)ActivatorUtilities.GetOrCreateInstance(_serviceProvider, converterType);
```